scripts/bulk-questions-uploader.js

```diff
@@ -0,0 +1,44 @@
+const fs = require("fs");
+const csv = require("csv-parser");
+const { createClient } = require("@supabase/supabase-js");
+const path = require("path");
+require("dotenv").config({ path: path.resolve(__dirname, "../.env") });
+
+const supabaseUrl = process.env.SUPABASE_URL;
+const supabaseKey = process.env.SUPABASE_KEY;
+
+if (!supabaseUrl || !supabaseKey) {
+    console.error("Please provide a SUPABASE_URL & SUPABASE_KEY in the .env file.");
+    process.exit(1);
+  }
+
+const supabase = createClient(supabaseUrl, supabaseKey);
+
+async function uploadData() {
+  try {
+    const results = [];
+    fs.createReadStream(path.resolve(__dirname, "questions.csv"))
+      .pipe(csv())
+      .on("data", (data) => results.push(data))
+      .on("end", async () => {
+        const records = results.map((row) => ({
+          question: row.question,
+          answers: [row.option_1, row.option_2, row.option_3, row.option_4],
+        }));
```

Comment:
instead of getting all rows into an array and then mapping through it, it would be more efficient in terms of memory to insert them on the handler of line 22. That way we avoid having an array that stores all rows (imagine it were a csv with millions of rows).

Reply:
resolving...